thoth-export-server/src/xml/onix3_google_books.rs

```diff
@@ -532,6 +532,10 @@ impl XmlElement<Onix3GoogleBooks> for ContributionType {
             ContributionType::INTRODUCTION_BY => "A24",
             ContributionType::AFTERWORD_BY => "A19",
             ContributionType::PREFACE_BY => "A15",
+            ContributionType::SOFTWARE_BY => "A30",
+            ContributionType::RESEARCH_BY => "A51",
+            ContributionType::CONTRIBUTIONS_BY => "A32",
+            ContributionType::INDEXER => "A34",
```

Comment:
A30, A34 and A51 are not supported by Google Books and should be treated the same way as Music Editor (below) - see [the list of supported values](https://support.google.com/books/partner/answer/6374180#zippy=%2Ccontributorrole-b-required).

Ideally we should also check whether or not the new types are accepted by the other formats. If they are not and we output them, they could lead to errors on ingest, or they may simply be dropped. In the latter case, it's misleading for us to keep including them, as we assume the data is being stored by the target platform when it's not.